### ISSUE TITLE
Allow "consistent" option for newline-per-chained-call

### ISSUE THREAD

**Issue by benblank** [eslint/eslint#10859]
**What rule do you want to change?** `newline-per-chained-call`

**Does this change cause the rule to produce more or fewer warnings?** fewer

**How will the change be implemented? (New option, new default behavior, etc.)?** new option

**Please provide some example code that this change will affect:**

```js
const one = foo().bar().baz().quux();

const two = foo()
    .bar()
    .baz()
    .quux();
```

**What does the rule currently do for this code?**  It produces warnings on the `const one` line, as there are no newlines between chained calls.

**What will the rule do after it's changed?**  When appropriately configured, it would allow *either* no newlines at all in the chain *or* a newline between each chained call, as long as the presence or absence of newlines is consistent within the chain.  This is similar to the `"consistent"` behavior for array-element-newline.

**Comment by yisibl** [comment]
We need this feature.
